### ISSUE TITLE
docs.python.org | 4.6. match Statements

### ISSUE THREAD

**Issue by ghost** [python/cpython#106780]
# Documentation

https://docs.python.org/3/tutorial/controlflow.html

I fixed the code as follows since points is not defined, still it does not work:

```python
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

points = [Point(0,7), Point(0,9)]

match points:
    case []:
        print("No points")
    case [Point(0, 0)]:
        print("The origin")
    case [Point(x, y)]:
        print(f"Single point {x}, {y}")
    case [Point(0, y1), Point(0, y2)]:
        print(f"Two on the Y axis at {y1}, {y2}")
    case _:
        print("Something else")
```

error:
```pytb
Traceback (most recent call last):
  File "c:\Users\moham\OneDrive\Education\UDST\gpaCal.py", line 15, in <module>
    case [Point(0, y1), Point(0, y2)]:
          ^^^^^^^^^^^^
TypeError: Point() accepts 0 positional sub-patterns (2 given)
```

### Linked PRs
* gh-106784
* gh-106819
* gh-106820

**Comment by terryjreedy** [comment]
That Point definition is in a previous example.  The paragraph that follows explains the solution for matching positional parameters.  In this case, add `    __match_args__ = ('x', 'y')` to Point and the example works.  I will augment the Point list example with a proper Point definition.  Working on PR.

**Comment by 60071203** [comment]
Dear Terry,

Many thanks for your prompt response.

I might be the oldest student -45yrs old - at my university (Uni. of Doha For Science and Tech = UDST), joined when I was 42yrs.

I had an OOP course in Fall22 - Java based -  where the instructor informed us, if we don't refer to original documentation, then we are not real programmers and we are not into it.

So, I have decided to even redo my Programming 1 and 2 classes in 2020/21, as I have relied on w3shools as a source also some of the slides provided by instructors at that period.

But, after my Java course, and learning how to read the official documentation of Java, I saw the depth, and now I see the difference between a programmer as a hobby and basic, and someone who is grasping the knowledge academically.

So, I have started reading your wonderful documentation on python.org, and I see how great efforts you and the team have placed to build such a treasure documentation covering all details from what the function, methods do and their impact on the ram, processing times, security, even how to write the code in best efficient ways with best algorithms.

Now, for the 4.6 match issue, and as a beginner,  i saw this part in the following "paragraph __match_args__ = ('x', 'y')" also this word "positional parameters" but I kept wondering, were these covered in chapters before 4.6. So I kept going back again to re-read the documentation form the beginning, and spent another like 4 hours, practicing and reading again all chapters before 4.6 but did not find things related to these two.  But I found a referenced link to classes, but I thought that would come after I complete chapter 4.6 and moving forward to next topics.

So maybe, it would be nice to consider the contents of the python.org documentation when there is a code that contains logics, methods, etc that will be covered later, to mention that, and to provide a tested workable code to avoid confusion for us as your students🌹

I will be also sharing with you some codes on the documentation also I see them maybe not clear in the aspired outputs, but I will do so in the bug reporting to not disturbing you a lot via emails.

Again, I thank you for the wonderful documentation for python.org, and I will share my wonderful experience with you to our College of Computing and Information Technology at my university, to ensure that we as students, not to learn from 3rd party, but to rely only on your official documentation for python.

Thank you again,
Mohammed AlMawlawi
A proud student at UDST

________________________________
